src/services/parser.service.ts

```diff
@@ -0,0 +1,14 @@
+import { parse, prepareParserConfig } from '../utils/parser';
+
+/**
+ * Service providing `@asyncapi/parser` functionality.
+ */
+export class ParserService {
+  public async parse(asyncapi: string | any, options?: ReturnType<typeof prepareParserConfig>) {
+    if (!asyncapi) {
+      return null;
+    }
+
+    return parse(asyncapi, options);
```

Comment:
I think a try catch block is needed here as `parse` method on the Parser-JS can throw an exception.

Reply:
yes, I didn't put the try catch on the `service` because I wasn't sure we could handle errors here. So we handle errors on the caller of the service: controller and middleware.
I also added on the PR description my decision process, but how do you think we should handle errors here?

Reply:
If you remove that service, these comments will be non relevant, am I right? 😅

Reply:
yes, thanks @magicmatatjahu 👍